src/test/java/org/products/productreviews/unittests/app/util/JDistanceTest.java

```diff
@@ -0,0 +1,129 @@
+     */
+    @Test
+    public void testJDistanceBothNoReviews() throws InvalidFormatException {
+        Account A = Account.createAccount(accountRepository, "User100", "Pass1234!");
+        Account B = Account.createAccount(accountRepository, "User101", "Pass1234!");
+
+        JDistance jDistanceAB = new JDistance(A, B);
+        float result = jDistanceAB.distance();
+        assertEquals(0, result);
+
+    }
+
+    /**
+     * Tests Jaccard distance if one account has no reviews
+     */
+    @Test
+    public void testJDistanceOneNoReviews() throws InvalidFormatException, InvalidKeyException {
+        Account A = Account.createAccount(accountRepository, "User100", "Pass1234!");
+        Account B = Account.createAccount(accountRepository, "User101", "Pass1234!");
+
+        Product example1 = Product.createProduct(productRepository,
+                "FountainPen",
+                399.99f,
+                "A Fountain Pen",
+                null);
+        Product example2 = Product.createProduct(productRepository,
+                "House",
+                10.39f,
+                "A House",
+                null);
+
+        Review rev1 = new Review(A, "Body", Review.Star.THREE);
+        Review rev2 = new Review(A, "Body", Review.Star.THREE);
+
+        rev1.setProduct(example1);
+        rev2.setProduct(example2);
+
+        A.addReview(rev1);
+        A.addReview(rev2);
+
+        JDistance jDistanceAB = new JDistance(A, B);
+        float result = jDistanceAB.distance();
+        assertEquals(0, result);
+
+    }
+
+    /**
+     * Tests Jaccard distance if both accounts have reviews, some matching some that are not.
+     */
+    @Test
+    public void testJDistanceBothReviews() throws InvalidFormatException, InvalidKeyException {
+        Account A = Account.createAccount(accountRepository, "User100", "Pass1234!");
+        Account B = Account.createAccount(accountRepository, "User101", "Pass1234!");
+
+        Product example1 = Product.createProduct(productRepository,
+                "FountainPen",
+                399.99f,
+                "A Fountain Pen",
+                null);
+        Product example2 = Product.createProduct(productRepository,
+                "House",
+                10.39f,
+                "A House",
+                null);
+
+        Review rev1 = new Review(A, "Body", Review.Star.THREE);
+        Review rev2 = new Review(A, "Body", Review.Star.THREE);
+
+        Review rev3 = new Review(B, "Body", Review.Star.TWO);
+        Review rev4 = new Review(B, "Body", Review.Star.THREE);
+
+        rev1.setProduct(example1);
+        rev3.setProduct(example1);
+
+        rev2.setProduct(example2);
+        rev4.setProduct(example2);
+
+        A.addReview(rev1);
+        A.addReview(rev2);
+
+        B.addReview(rev3);
+        B.addReview(rev4);
+
+        JDistance jDistanceAB = new JDistance(A, B);
+        float result = jDistanceAB.distance();
+        assertEquals(1f/3f, result);
+
+    }
```

Comment:
Maybe a test with 3 accounts? Just to test if the "similar" account actually has a smaller distance.

Reply:
Added in [5484212](https://github.com/TrustyUnitReview/ProductReview/pull/87/commits/54842126deb8f3ad0b6c89496025c7f60460c466)